Client/Frontend/Settings/HomepageSettings/WallpaperSettings/v2/WallpaperSettingsViewController.swift

```diff
@@ -0,0 +1,285 @@
+                           observing: [.DisplayThemeChanged, UIContentSizeCategory.didChangeNotification])
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        applyTheme()
+
+        collectionView.selectItem(at: viewModel.selectedIndexPath, animated: false, scrollPosition: [])
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass
+            || previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            updateOnRotation()
+        }
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        updateOnRotation()
+    }
+}
+
+// MARK: - CollectionView Data Source
+extension WallpaperSettingsViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return viewModel.numberOfSections
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return viewModel.numberOfWallpapers(in: section)
+    }
+
+    func collectionView(_ collectionView: UICollectionView,
+                        viewForSupplementaryElementOfKind kind: String,
+                        at indexPath: IndexPath) -> UICollectionReusableView {
+        guard kind == UICollectionView.elementKindSectionHeader,
+              let headerView = collectionView.dequeueReusableSupplementaryView(
+                ofKind: UICollectionView.elementKindSectionHeader,
+                withReuseIdentifier: WallpaperSettingsHeaderView.cellIdentifier,
+                for: indexPath) as? WallpaperSettingsHeaderView,
+              let headerViewModel = viewModel.sectionHeaderViewModel(for: indexPath.section, dismissView: {
+                  self.dismissView()
+              })
+        else { return UICollectionReusableView() }
+
+        headerView.configure(viewModel: headerViewModel)
+        return headerView
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: WallpaperCollectionViewCell.cellIdentifier,
+            for: indexPath) as? WallpaperCollectionViewCell,
+                let cellViewModel = viewModel.cellViewModel(for: indexPath)
+        else { return UICollectionViewCell() }
+
+        cell.viewModel = cellViewModel
+        return cell
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        activityIndicatorView.startAnimating()
```

Comment:
SHould we have a minor comment here, explaining that, basically, we always download wallpapers? when selecting stuff, which is why there's no check for whether or not we should `startAnimating()`? I only ask because I saw this and thought, "Shouldn't we check if it exists before we start animating?" And then I remember why.

And I'm working on wallpapers right now! lol